internal/services/vehicle_signals_event_property_service.go

```diff
@@ -45,6 +45,27 @@ func (v *vehicleSignalsEventPropertyService) GenerateData(ctx context.Context, d
 
 	for key, value := range eventAvailableProperties {
 
+		// Validate if value is required.
+		if _, ok := data[key]; ok {
```

Comment:
what if we put this in a function below eg. called `shouldValueBeConsidered` - if false, continue loop to skip current.